src/jit/gentree.cpp

```diff
@@ -8826,6 +8830,13 @@ unsigned GenTree::NumChildren()
         {
             return 3;
         }
+#if FEATURE_HW_INTRINSICS
+        // TODO-XArch-CQ: Check for possibility of setting SetZero or similar intrinsics to unary operator format
+        if ((OperGet() == GT_HWIntrinsic) && (AsHWIntrinsic()->gtHWIntrinsicId == NI_SSE2_SetZeroVector128))
```

Comment:
Why do we need this?

Reply:
Yeah. The zero operator case should already be handled.... Its working for `Sse.SetZero`.

Wondering what the difference is here.

Reply:
I guess that it could be leftover from my old implementation - will remove if not needed anymore.

Reply:
Unfortunately I need to keep it. After removing this fragment I am getting the following error:
```
E:\src\ms\dotnet\coreclr-sse2\bin\tests\Windows_NT.x64.Checked\TestWrappers\JIT.HardwareIntrinsics\JIT.HardwareIntrinsics.XUnitWrapper.cs(5427): error : JIT_HardwareIntrinsics._X86_Sse2_SetZeroVector1
28_ro_SetZeroVector128_ro_._X86_Sse2_SetZeroVector128_ro_SetZeroVector128_ro_cmd [FAIL] [E:\src\ms\dotnet\coreclr-sse2\tests\runtest.proj]

        Assert failure(PID 12640 [0x00003160], Thread: 4688 [0x1250]): Assertion failed 'gtOp.gtOp1 != nullptr' in 'IntelHardwareIntrinsicTest.Program:Main(ref):int' (IL size 1116)

            File: e:\src\ms\dotnet\coreclr-sse2\src\jit\gentree.cpp Line: 8833
            Image: E:\src\ms\dotnet\coreclr-sse2\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe



  Return code:      1
  Raw output file:      E:\src\ms\dotnet\coreclr-sse2\bin\tests\Windows_NT.x64.Checked\Reports\JIT.HardwareIntrinsics\X86\Sse2\SetZeroVector128_ro\SetZeroVector128_ro.output.txt
  Raw output:
  BEGIN EXECUTION
         "E:\src\ms\dotnet\coreclr-sse2\tests\..\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" SetZeroVector128_ro.exe
        Expected: 100
        Actual: 123456789
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1

  To run the test:
  > set CORE_ROOT=E:\src\ms\dotnet\coreclr-sse2\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root
  > E:\src\ms\dotnet\coreclr-sse2\bin\tests\Windows_NT.x64.Checked\JIT\HardwareIntrinsics\X86\Sse2\SetZeroVector128_ro\SetZeroVector128_ro.cmd

        Expected: True
        Actual:   False
          E:\src\ms\dotnet\coreclr-sse2\bin\tests\Windows_NT.x64.Checked\TestWrappers\JIT.HardwareIntrinsics\JIT.HardwareIntrinsics.XUnitWrapper.cs(5427,0): at JIT_HardwareIntrinsics._X86_Sse2_SetZero
  Vector128_ro_SetZeroVector128_ro_._X86_Sse2_SetZeroVector128_ro_SetZeroVector128_ro_cmd()
    Finished:    JIT.HardwareIntrinsics.XUnitWrapper
  === TEST EXECUTION SUMMARY ===
     JIT.HardwareIntrinsics.XUnitWrapper  Total: 258, Errors: 0, Failed: 1, Skipped: 0, Time: 53,861s

Build FAILED.

E:\src\ms\dotnet\coreclr-sse2\bin\tests\Windows_NT.x64.Checked\TestWrappers\JIT.HardwareIntrinsics\JIT.HardwareIntrinsics.XUnitWrapper.cs(5427): error : JIT_HardwareIntrinsics._X86_Sse2_SetZeroVector1
28_ro_SetZeroVector128_ro_._X86_Sse2_SetZeroVector128_ro_SetZeroVector128_ro_cmd [FAIL] [E:\src\ms\dotnet\coreclr-sse2\tests\runtest.proj]
    0 Warning(s)
    1 Error(s)
```